CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+# Copyright 2022 The Foundry Visionmongers Ltd
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.21)
+
+project(OpenAssetIO-Test-CMake)
+
+set(CMAKE_CXX_STANDARD 17)
+enable_testing()
+
+#-----------------------------------------------------------------------
+# Options
+
+option(OPENASSETIOTEST_ENABLE_SUBPROJECT "Use OpenAssetIO as a subproject" OFF)
+option(OPENASSETIOTEST_ENABLE_PYTHON "Test Python bindings" ON)
+option(OPENASSETIOTEST_ENABLE_C "Test C bindings" ON)
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+    option(OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI" OFF)
+endif ()
+
+if (CMAKE_TOOLCHAIN_FILE)
+    message(STATUS "Using toolchain file                = ${CMAKE_TOOLCHAIN_FILE}")
+endif ()
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    message(STATUS "New C++11 ABI for GCC               = ${OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI}")
+endif ()
+message(STATUS "Testing OpenAssetIO as a subproject = ${OPENASSETIOTEST_ENABLE_SUBPROJECT}")
+message(STATUS "Testing OpenAssetIO Python bindings = ${OPENASSETIOTEST_ENABLE_PYTHON}")
+message(STATUS "Testing OpenAssetIO C bindings      = ${OPENASSETIOTEST_ENABLE_C}")
+
+
+#-----------------------------------------------------------------------
+# OpenAssetIO dependency.
+
+if (OPENASSETIOTEST_ENABLE_SUBPROJECT)
+    # Use OpenAssetIO as a subproject.
+    #
+    # This will add to the project using `add_subdirectory`.
+    #
+    # OpenAssetIO must therefore be built, meaning its `find_package`
+    # calls for dependencies must work.
+
+    add_subdirectory(OpenAssetIO)
+else ()
+    # Use OpenAssetIO as an external CMake package
+    #
+    # The `OpenAssetIOConfig.cmake` file must be discoverable. E.g.
+    # add the install directory (`dist`) to CMAKE_PREFIX_PATH.
+
+    find_package(OpenAssetIO REQUIRED)
```

Comment:
Should you always find OpenAssetIO by CMake config file? I assume so, as it's the modern way.

Realise that CMake will eventually find it with the correct search paths set up, but wondering whether it's better to advise this on a public test, because readers might copy-n-paste.

Reply:
Unfortunately not for the case of public ConanCenter packages, according to [their docs](https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#package)

> CMake config files must be removed (they will be generated for consumers by cmake_find_package, cmake_find_package_multi or CMakeDeps generators).

The (eventual) public conanfile will have to copy the `OpenAssetIOConfig` logic into a generated `FindOpenAssetIO` module :disappointed:

Reply:
Well, this is interesting, because I was on the Conan 2 hackathon call this week, and it was brought up that packages can provide their own CMake config files, and you can describe in receipe when they do. I even asked a question on it!

Reply:
Maybe this is a ConanCenter-specific policy? Or perhaps the policy will change with Conan 2?

Reply:
My impression was that it's in the latest Conan 1.x too. But maybe it'll change. Anyway, you need to be compatible with their policies as they are to get through a PR to Conan Center.